engine/src/flutter/lib/ui/semantics/semantics_update_builder_unittests.cc

```diff
@@ -88,5 +88,81 @@ TEST_F(SemanticsUpdateBuilderTest, CanHandleAttributedStrings) {
+
+  auto nativeSemanticsUpdate = [message_latch](Dart_NativeArguments args) {
+    auto handle = Dart_GetNativeArgument(args, 0);
+    intptr_t peer = 0;
+    Dart_Handle result = Dart_GetNativeInstanceField(
+        handle, tonic::DartWrappable::kPeerIndex, &peer);
+    ASSERT_FALSE(Dart_IsError(result));
+    SemanticsUpdate* update = reinterpret_cast<SemanticsUpdate*>(peer);
+    SemanticsNodeUpdates nodes = update->takeNodes();
+    ASSERT_EQ(nodes.size(), (size_t)1);
+    auto node = nodes.find(0)->second;
+    // Should match the updateNode in ui_test.dart.
+    ASSERT_EQ(node.label, "label");
+    ASSERT_EQ(node.labelAttributes.size(), (size_t)1);
+    ASSERT_EQ(node.labelAttributes[0]->start, 1);
+    ASSERT_EQ(node.labelAttributes[0]->end, 2);
+    ASSERT_EQ(node.labelAttributes[0]->type, StringAttributeType::kSpellOut);
+
+    ASSERT_EQ(node.value, "value");
+    ASSERT_EQ(node.valueAttributes.size(), (size_t)1);
+    ASSERT_EQ(node.valueAttributes[0]->start, 2);
+    ASSERT_EQ(node.valueAttributes[0]->end, 3);
+    ASSERT_EQ(node.valueAttributes[0]->type, StringAttributeType::kSpellOut);
+
+    ASSERT_EQ(node.hint, "hint");
+    ASSERT_EQ(node.hintAttributes.size(), (size_t)1);
+    ASSERT_EQ(node.hintAttributes[0]->start, 0);
+    ASSERT_EQ(node.hintAttributes[0]->end, 1);
+    ASSERT_EQ(node.hintAttributes[0]->type, StringAttributeType::kLocale);
+    auto local_attribute =
+        std::static_pointer_cast<LocaleStringAttribute>(node.hintAttributes[0]);
+    ASSERT_EQ(local_attribute->locale, "en-MX");
+
+    ASSERT_EQ(node.increasedValue, "increasedValue");
+    ASSERT_EQ(node.increasedValueAttributes.size(), (size_t)1);
+    ASSERT_EQ(node.increasedValueAttributes[0]->start, 4);
+    ASSERT_EQ(node.increasedValueAttributes[0]->end, 5);
+    ASSERT_EQ(node.increasedValueAttributes[0]->type,
+              StringAttributeType::kSpellOut);
+
+    ASSERT_EQ(node.decreasedValue, "decreasedValue");
+    ASSERT_EQ(node.decreasedValueAttributes.size(), (size_t)1);
+    ASSERT_EQ(node.decreasedValueAttributes[0]->start, 5);
+    ASSERT_EQ(node.decreasedValueAttributes[0]->end, 6);
+    ASSERT_EQ(node.decreasedValueAttributes[0]->type,
+              StringAttributeType::kSpellOut);
+    message_latch->Signal();
+  };
+
+  Settings settings = CreateSettingsForFixture();
+  TaskRunners task_runners("test",                  // label
+                           GetCurrentTaskRunner(),  // platform
+                           CreateNewThread(),       // raster
+                           CreateNewThread(),       // ui
+                           CreateNewThread()        // io
+  );
+
+  AddNativeCallback("SemanticsUpdate",
+                    CREATE_NATIVE_ENTRY(nativeSemanticsUpdate));
+
+  std::unique_ptr<Shell> shell = CreateShell(settings, task_runners);
+
+  ASSERT_TRUE(shell->IsSetup());
+  auto configuration = RunConfiguration::InferFromSettings(settings);
+  configuration.SetEntrypoint("sendSemanticsUpdate");
```

Comment:
Is this where you meant to call `sendSemanticsUpdateWithRole`?

Reply:
oh yeah, I forgot to change it back